features/projects/types/project.types.ts

```diff
@@ -1,31 +1,31 @@
 export enum ProjectLanguage {
-	react = 'react',
-	node = 'node',
-	python = 'python',
+  react = "react",
+  node = "node",
+  python = "python",
 }
 
 export enum ProjectStatus {
-	stable = 'stable',
-	warning = 'warning',
-	critical = 'critical',
-	error = 'error',
-	info = 'info',
+  stable = "stable",
+  warning = "warning",
+  critical = "critical",
+  error = "error",
+  info = "info",
```

Comment:
In my opinion, the enum should reflect the possible values in the API data. So I'd rather replace `stable` and `critical` with `info` and `error` instead of adding them. After all, `stable` and `critical` don't exist in the API data.

Can you think of a solution using this approach?

Reply:
I think an approach would be to not update the data in the use-projects data fetch hook, and instead update values somewhere in the components in the projects feature. This will probably solve the issue of the failing cypress test that consumes the getProjects data.

Reply:
I recommend not mutating the data at all. That's usually bad practice because it could unknowingly affect other places as well. E.g. let's imagine an object is passed down a component tree as a prop and one of the children mutates it like

```
if (project.status === "error") {
  project.status = "critical"
}
```

If it is passed down the component tree as a reference `projects.status` will have been changed everywhere. Potentially that would affect another seemingly unrelated component. Such a bug would be very hard to detect.

You can however use a very simple function that returns the text to be displayed based on the status

```
function getStatusText(status: ProjectStatus) {
  ...
}
```

Does that make sense?

Reply:
I appreciate all the feedback. I think that makes sense. I like to believe I just took a much better approach to solve this in a new branch I made(named it project-list-doesnt-show-correct-status) as it felt more appropriate than this one. I coded it like this, project-card:
`
const statusColors = {
  [ProjectStatus.info]: BadgeColor.success,
  [ProjectStatus.warning]: BadgeColor.warning,
  [ProjectStatus.error]: BadgeColor.error,
};

const statusText = {
  [ProjectStatus.info]: "stable",
  [ProjectStatus.warning]: "warning",
  [ProjectStatus.error]: "critical",
};
`
Then in the JSX:
`
            <Badge color={statusColors[status]}>
              {capitalize(statusText[status])}
            </Badge>
`
I hope that's better

Reply:
Here's the link to the branch: https://github.com/profydev/prolog-app-ari7946/pull/13
I must have merged in within minutes of seeing your recent comment 😅 , otherwise I would have left it open for a bit

Reply:
Yeah that's exactly how I would do it. You can even get rid of the `capitalize` function of you adjust the strings